From 2d626af27e8f3115175749d183a8b7f91849bf10 Mon Sep 17 00:00:00 2001
From: Joachim Wiberg <troglobit@gmail.com>
Date: Sun, 14 Jul 2024 17:43:39 +0200
Subject: [PATCH 06/17] Fix annoying gcc warning in rotate_file()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Addiva Elektronik

syslogd.c: In function ‘rotate_file’:
syslogd.c:1787:69: warning: ‘%s’ directive output between 0 and 2147483646 bytes may cause result to exceed ‘INT_MAX’ [-Wformat-truncation=]
 1787 |                                 snprintf(cmd, sizeof(cmd), "gzip -f %s", newFile);
      |                                                                     ^~

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 src/syslogd.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/syslogd.c b/src/syslogd.c
index c838c36..461bbb1 100644
--- a/src/syslogd.c
+++ b/src/syslogd.c
@@ -1786,10 +1786,11 @@ static void rotate_file(struct filed *f, struct stat *stp_or_null)
 			snprintf(newFile, len, "%s.%d", f->f_un.f_fname, i);
 
 			if (!rename(oldFile, newFile) && i > 0) {
-				size_t clen = 18 + strlen(newFile) + 1;
+				const char *gzip = "gzip -f";
+				size_t clen = strlen(gzip) + len + 1;
 				char cmd[clen];
 
-				snprintf(cmd, sizeof(cmd), "gzip -f %s", newFile);
+				snprintf(cmd, clen, "%s %s", gzip, newFile);
 				system(cmd);
 			}
 		}
-- 
2.43.0

